mongodb/src/main/java/org/geotools/data/mongodb/MongoInferredMapper.java

```diff
@@ -80,6 +84,18 @@ public SimpleFeatureType buildFeatureType(Name name, DBCollection collection) {
             }
         }
         
+        //Examine the DBO and remove any invalid indexed fields (such as arrays)
+        DBObject dbo = collection.findOne();
+        if (dbo != null) {
+        	Iterator<String> indexedIterator = indexedFields.iterator();
+        	while (indexedIterator.hasNext()) {
+        		Object value = MongoUtil.getDBOValue(dbo, indexedIterator.next());
+        		if (value == null) {
+        			indexedIterator.remove();
+        		}
+        	}
+        }
+        
```

Comment:
You may want to add a javadoc note to yourself; this appears to be where the collection is scanned to generate a FeatureType.

We have some kind of ContentState refresh method that can be used to ask a DataStore to "rescan".